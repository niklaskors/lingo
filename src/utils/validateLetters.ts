import { Validation } from '../view/Game';

export function validateLetters(chosenWord: string, correctWord:string) {
  const letters = chosenWord.split('');
  const validation: Validation[] =[];

  for(let i = 0; i < letters.length; i++) {
    const letter = letters[i];
    let status: Validation['status'] = 'absent';

    if (!correctWord.includes(letter)) {
      status = 'absent';
    } else if (correctWord[i] === letter) {
      status = 'correct';
    } else if (correctWord.includes(letter)) {
      status = 'present';
    }

    validation.push({
      letter,
      status
    })
  }

    return validation;
  }