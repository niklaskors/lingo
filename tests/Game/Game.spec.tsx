import { validateLetters } from '../../src/utils/validateLetters';

test('should validate letters correctly', () => {
  const correctWord = 'energy';
  const chosenWord = 'garden';

  const validation = validateLetters(chosenWord, correctWord);
  expect(validation).toMatchSnapshot();
})

test('should validate letters correctly', () => {
  const correctWord = 'energy';
  const chosenWord = 'eeeeee';

  const validation = validateLetters(chosenWord, correctWord);
  expect(validation).toMatchSnapshot();
})

test('should validate letters correctly', () => {
  const correctWord = 'energy';
  const chosenWord = 'energn';

  const validation = validateLetters(chosenWord, correctWord);
  expect(validation).toMatchSnapshot();
})