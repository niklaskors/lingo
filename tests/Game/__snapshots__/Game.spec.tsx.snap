// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should validate letters correctly 1`] = `
[
  {
    "letter": "g",
    "status": "present",
  },
  {
    "letter": "a",
    "status": "absent",
  },
  {
    "letter": "r",
    "status": "present",
  },
  {
    "letter": "d",
    "status": "absent",
  },
  {
    "letter": "e",
    "status": "present",
  },
  {
    "letter": "n",
    "status": "present",
  },
]
`;

exports[`should validate letters correctly 2`] = `
[
  {
    "letter": "e",
    "status": "correct",
  },
  {
    "letter": "e",
    "status": "absent",
  },
  {
    "letter": "e",
    "status": "correct",
  },
  {
    "letter": "e",
    "status": "absent",
  },
  {
    "letter": "e",
    "status": "absent",
  },
  {
    "letter": "e",
    "status": "absent",
  },
]
`;

exports[`should validate letters correctly 3`] = `
[
  {
    "letter": "e",
    "status": "correct",
  },
  {
    "letter": "n",
    "status": "correct",
  },
  {
    "letter": "e",
    "status": "absent",
  },
  {
    "letter": "r",
    "status": "correct",
  },
  {
    "letter": "g",
    "status": "correct",
  },
  {
    "letter": "n",
    "status": "absent",
  },
]
`;
